kolibri/plugins/coach_tools/assets/src/actions.js

```diff
@@ -14,82 +24,121 @@ function redirectToDefaultReports(store, params) {
   store.dispatch('CORE_SET_PAGE_LOADING', true);
   store.dispatch('SET_PAGE_NAME', 'REPORTS_NO_QUERY');
 
-  // if necessary query server root topic PK
-  // TODO: Get channel id, root id, and facility id.
-  const channelId = 'channel_id';
-  const contentScopeId = 'root_id';
-  const userScopeId = 'facility_id';
-
-  router.replace({
-    name: PageNames.REPORTS,
-    params: {
-      channel_id: channelId,
-      content_scope: ReportsOptions.CONTENT_SCOPE_OPTIONS[0],
-      content_scope_id: contentScopeId,
-      user_scope: ReportsOptions.USER_SCOPE_OPTIONS[0],
-      user_scope_id: userScopeId,
-      all_or_recent: ReportsOptions.ALL_OR_RECENT_OPTIONS[0],
-      view_by_content_or_learners: ReportsOptions.VIEW_BY_CONTENT_OR_LEARNERS_OPTIONS[0],
-      sort_column: ReportsOptions.SORT_COLUMN_OPTIONS[0],
-      sort_order: ReportsOptions.SORT_ORDER_OPTIONS[0],
+  const channelListPromise = ChannelResource.getCollection({}).fetch();
+  const facilityIdPromise = FacilityUserResource.getCurrentFacility();
+
+  ConditionalPromise.all([channelListPromise, facilityIdPromise]).only(
+    samePageCheckGenerator(store),
+    ([channelList, facilityId]) => {
+      /* get current channelId */
+      const channelId = getDefaultChannelId(channelList);
+
+      /* get contentScopeId for root */
+      let contentScopeId = null;
+      for (let x = 0; x < channelList.length; x++) {
+        if (channelList[x].id === channelId) {
+          contentScopeId = channelList[x].root_pk;
+        }
+      }
+
+      /* get userScopeId for facility */
+      const userScopeId = facilityId[0];
+      router.replace({
+        name: PageNames.REPORTS,
+        params: {
+          channel_id: channelId,
+          content_scope: ReportsOptions.CONTENT_SCOPE_OPTIONS[0],
+          content_scope_id: contentScopeId,
+          user_scope: ReportsOptions.USER_SCOPE_OPTIONS[0],
+          user_scope_id: userScopeId,
+          all_or_recent: ReportsOptions.ALL_OR_RECENT_OPTIONS[0],
+          view_by_content_or_learners: ReportsOptions.VIEW_BY_CONTENT_OR_LEARNERS_OPTIONS[0],
+          sort_column: ReportsOptions.SORT_COLUMN_OPTIONS[0],
+          sort_order: ReportsOptions.SORT_ORDER_OPTIONS[0],
+        },
+      });
     },
-  });
-}
 
+    error => {
+      coreActions.handleError(store, error);
+    }
+  );
+}
 
 function showReports(store, params) {
-  store.dispatch('CORE_SET_PAGE_LOADING', true);
+  store.dispatch('CORE_SET_PAGE_LOADING', true); // does this even work since I didn't implement it?
```

Comment:
The `core-base` component is supposed to use this variable to hide the main `content` slot when `loading === true`